packages/js/product-editor/src/components/header/header.tsx

```diff
@@ -1,14 +1,46 @@
 /**
  * External dependencies
  */
-import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
 import { createElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Product } from '@woocommerce/data';
 
 export type HeaderProps = {
+	product?: Product;
 	title: string;
 };
 
-export function Header( { title }: HeaderProps ) {
+export function Header( { product, title }: HeaderProps ) {
+	const { isProductLocked, isSaving } = useSelect(
+		( select ) => {
+			const { isSavingEntityRecord } = select( 'core' );
+			const { isPostSavingLocked } = select( 'core/editor' );
+			return {
+				// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+				// @ts-ignore
+				isProductLocked: isPostSavingLocked(),
+				// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+				// @ts-ignore
+				isSaving: isSavingEntityRecord(
+					'postType',
+					'product',
+					product?.id
+				),
+			};
+		},
+		[ product?.id ]
+	);
+
+	const isDisabled = isProductLocked || isSaving;
+
+	const { saveEditedEntityRecord } = useDispatch( 'core' );
+
+	function handleSave() {
+		saveEditedEntityRecord( 'postType', 'product', product?.id );
```

Comment:
Optional, but wondering if we will always have the `product` or need to use `?` to check its existence.

Reply:
Done -> [3686408](https://github.com/woocommerce/woocommerce/pull/37152/commits/3686408957cc875e5336470df5b8da1d5c8beadd)